src/jit/ByteCodeParser.cpp

```diff
@@ -1907,24 +1907,27 @@ static void compileFunction(JITCompiler* compiler)
             }
             break;
         }
-        case ByteCode::I32AtomicRmwAddOpcode:
         case ByteCode::I32AtomicRmw8AddUOpcode:
         case ByteCode::I32AtomicRmw16AddUOpcode:
-        case ByteCode::I32AtomicRmwSubOpcode:
         case ByteCode::I32AtomicRmw8SubUOpcode:
         case ByteCode::I32AtomicRmw16SubUOpcode:
-        case ByteCode::I32AtomicRmwAndOpcode:
         case ByteCode::I32AtomicRmw8AndUOpcode:
         case ByteCode::I32AtomicRmw16AndUOpcode:
-        case ByteCode::I32AtomicRmwOrOpcode:
         case ByteCode::I32AtomicRmw8OrUOpcode:
         case ByteCode::I32AtomicRmw16OrUOpcode:
-        case ByteCode::I32AtomicRmwXorOpcode:
         case ByteCode::I32AtomicRmw8XorUOpcode:
         case ByteCode::I32AtomicRmw16XorUOpcode:
-        case ByteCode::I32AtomicRmwXchgOpcode:
         case ByteCode::I32AtomicRmw8XchgUOpcode:
         case ByteCode::I32AtomicRmw16XchgUOpcode: {
+            compiler->increaseStackTmpSize(16);
+            FALLTHROUGH;
+        }
+        case ByteCode::I32AtomicRmwAddOpcode:
+        case ByteCode::I32AtomicRmwSubOpcode:
+        case ByteCode::I32AtomicRmwAndOpcode:
+        case ByteCode::I32AtomicRmwOrOpcode:
+        case ByteCode::I32AtomicRmwXorOpcode:
+        case ByteCode::I32AtomicRmwXchgOpcode: {
             info = Instruction::kIs32Bit;
             requiredInit = OTAtomicRmwI32;
```

Comment:
There should be an `OTAtomicRmwShort` which adds an extra tmp register when short atomic is not present. Could use the compiler option bits to check this.